common/workunit/workunit.cpp

```diff
@@ -9002,6 +9001,25 @@ void CLocalWorkUnit::setStatistic(StatisticCreatorType creatorType, const char *
     }
     if (kind == StCostCompile)
         p->setPropInt64("@costCompile", value);
+    // Special case - update aggregates for dfu FileAccessCost
+    // This is necessary because stats written to global stats are not aggregated automatically.
+    // Note, 1) StatisticAggregator is only executed by graphs but fileservices are execute outside
+    // graphs.  This means dfu FileAccessCost will not be processed in StatisticAggregator unless
+    // a graph in the same workflow as the dfu workunit is executed subsequently. 2) Note,
+    // that the only stats that are "dirty" in StatisticAggregator are ones that have been loaded
+    // with IStatisticCollection::recordStats.
+    if (scopeType == SSTdfuworkunit && kind == StCostFileAccess && value)
+    {
+        StringBuffer currentScope(scope), parent;
+        while (getParentScope(parent.clear(), currentScope.str()))
+        {
+            currentScope.set(parent);
+            StatisticScopeType sst = getScopeType(queryScopeTail(currentScope.str()));
+            if (sst!=SSTnone)
+                setStatistic(creatorType, creator, sst, currentScope.str(), StCostFileAccess, "", value, 1, 0, StatsMergeSum);
+        }
+        setStatistic(creatorType, creator, SSTglobal, "", StCostFileAccess, "", value, 1, 0, StatsMergeSum);
```

Comment:
Discussed offline, this would cause problems if called >once for same scope. This will happen if the duration if more than a few seconds, these mergesum figures will be wrong.